arangosh/Shell/V8ShellFeature.cpp

```diff
@@ -511,6 +511,33 @@ int V8ShellFeature::runShell(std::vector<std::string> const& positionals) {
       input = "help()";
     }
 
+    bool foundShellTool = false;
+    if (i.length() > 0 && i[0] == 'a') {
+      auto pos = i.find(' ');
+      if (pos != std::string::npos) {
+        // is it a variable name assignment?
+        if (i.find('=', pos) == std::string::npos) {
+          i = i.substr(0, pos);
+        }
+      }
+      if (i == "arangobackup" ||
+          i == "arangobench" ||
+          i == "arangod" ||
+          i == "arangodbtests" ||
+          i == "arangodump" ||
+          i == "arangoexport" ||
+          i == "arangoimp" ||
+          i == "arangoimport" ||
+          i == "arango-init-database" ||
+          i == "arangoinspect" ||
+          i == "arangorestore" ||
+          i == "arango-secure-installation" ||
+          i == "arangosh" ||
+          i == "arangovpack"
+          ) {
+        foundShellTool = true;
+      }
```

Comment:
All this should be moved into a function and then be called when we are already sure that an error has occurred.
There is no need to run this check for every command executed.